### PR TITLE
Ps protein dna annotation

### PR DESCRIPTION
**Pull request by preetisi**

### CODE REVIEW

File: altanalyze3/components/annotation/ensembl_biomart.py [hunk 1]
```diff
@@ -0,0 +1,503 @@
+"""
+This is a generalized python module for getting data from Ensemble using Biomart server.
+"""
+
+from __future__ import absolute_import, division, print_function
```
Comment (michael-kotliar) [inline altanalyze3/components/annotation/ensembl_biomart.py:5]: Are these functions for some kind of backward compatibility with Python 2.7?

File: altanalyze3/components/annotation/ensembl_biomart.py [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+import requests
+
+from future.utils import native_str
+from builtins import *
```
Comment (michael-kotliar) [inline altanalyze3/components/annotation/ensembl_biomart.py:9]: import * is usually not recommended in Python

File: altanalyze3/components/annotation/ensembl_biomart.py [hunk 3]
```diff
@@ -0,0 +1,503 @@
...
+    def __init__(self, host=None, path=None, port=None):
+        """ServerBase constructor.
+        Args:
+            host (str): Url of host to connect to.
+            path (str): Path on the host to access to the biomart service.
+            port (int): Port to use for the connection.
+            use_cache (bool): Whether to cache requests.
+        """
+        # Use defaults if arg is None.
+        host = host or DEFAULT_HOST
```
Comment (michael-kotliar) [inline altanalyze3/components/annotation/ensembl_biomart.py:41]: If you use None as the default value, it's better to check against None
`host = DEFAULT_HOST if host is None else host`

File: altanalyze3/components/annotation/ensembl_biomart.py [hunk 4]
```diff
@@ -0,0 +1,503 @@
...
+        if not url.startswith('http://') or url.startswith('https://'):
+            url = prefix + url
+        return url
+
+    @staticmethod
+    def _remove_trailing_slash(url):
+        if url.endswith('/'):
+            url = url[:-1]
+        return url
+
```
Comment (michael-kotliar) [inline altanalyze3/components/annotation/ensembl_biomart.py:88]: You can use `strip` for that
```
>>> a = "line/"
>>> a.strip("/")
'line'
>>> a = "line"
>>> a.strip("/")
'line'
>>> 
```

File: altanalyze3/components/annotation/ensembl_biomart.py (hunk 2 above)
Comment (preetisi) [reply altanalyze3/components/annotation/ensembl_biomart.py:9]: Agreed.

File: altanalyze3/components/annotation/ensembl_biomart.py (hunk 4 above)
Comment (preetisi) [reply altanalyze3/components/annotation/ensembl_biomart.py:88]: good suggestion

File: altanalyze3/components/annotation/ensembl_biomart.py (hunk 3 above)
Comment (michael-kotliar) [reply altanalyze3/components/annotation/ensembl_biomart.py:41]: Yep, but when you call this function without `host` parameter provided the default value will be None.

https://python-patterns.guide/python/sentinel-object/

**Pull request merged**